tests/unittests/test_backoff.py

```diff
@@ -287,21 +337,28 @@ def test_connection_reset_error_occurred__list_subscriber(self, mocked_get, mock
         self.assertEquals(mocked_get.call_count, 5)
 
     @mock.patch("FuelSDK.rest.ET_GetSupport.get")
-    def test_connection_reset_error_occurred__list_subscriber__get_all_subscribers_list(self, mocked_get, mocked_sleep):
+    @mock.patch("tap_exacttarget.endpoints.list_subscribers.ListSubscriberDataAccessObject._get_all_subscribers_list")
+    def test_connection_reset_error_occurred__list_subscriber__get_all_subscribers_list(self, mocked_get_all_subscribers_list, mocked_get, mocked_sleep):
+        """
+            Test case to verify that we backoff for 5 times when 'ConnectionResetError' error occurs
+        """
         # mock 'get' and raise error
         mocked_get.side_effect = socket.error(104, 'Connection reset by peer')
         # make the object of 'ListSubscriberDataAccessObject'
         obj = list_subscribers.ListSubscriberDataAccessObject({"start_date": "2020-01-01T00:00:00Z"}, {}, None, {})
         try:
             # call function
-            obj._get_all_subscribers_list()
+            obj.sync_data()
```

Comment:
Can you explain this code change?

Reply:
We have added backoff on `sync_data`, as the function `_get_all_subscribers_list` will be called from that function and if we add backoff on the **_get_all_subscribers_list**, there are chances of the Tap backing off 25 times.